arbnode/dataposter/dataposter_test.go

```diff
@@ -242,3 +243,27 @@ func (s *server) mux(w http.ResponseWriter, r *http.Request) {
 		fmt.Printf("error writing response: %v\n", err)
 	}
 }
+
+func TestMaxFeeCapFormulaCalculation(t *testing.T) {
```

Comment:
This test takes evaluation expression and plugs parameters that dataposter is supposed to plug and evaluates it. This doesn't really test that dataposter will actually plug those parameters during evaluation.

If you factor out eval function into helper as I suggested above, then you can initialize dataposter struct with some config, pass backlogOfBatches and see if it evaluates correctly.

Reply:
Agreed.